kyaml/fn/runtime/container/container.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package container
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	runtimeexec "sigs.k8s.io/kustomize/kyaml/fn/runtime/exec"
+	"sigs.k8s.io/kustomize/kyaml/fn/runtime/runtimeutil"
+
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Filter filters Resources using a container image.
+// The container must start a process that reads the list of
+// input Resources from stdin, reads the Configuration from the env
+// API_CONFIG, and writes the filtered Resources to stdout.
+// If there is a error or validation failure, the process must exit
+// non-zero.
+// The full set of environment variables from the parent process
+// are passed to the container.
+//
+// Function Scoping:
```

Comment:
Kudos for writing all this out.

_Might_ be cleaner to  add a ptr to here to the package that enforces scope,
and expound on details there (few people who's initials aren't JL will care).

Here just show the simple case and be clear
on how stdin, stderr, stdout, env vars etc. get set up.

Unless these rules somehow drive the use cases discussed, in which i guess it has to be here.

kustomize also has rules and tests covering kustomization (transformer) scope.
https://github.com/kubernetes-sigs/kustomize/blob/master/api/loader/fileloader.go
That loader cannot be used here, because the models are too diverged now (overlays trees are just different from all this)
but i recall it being tricky to write and cover with tests, and it pushed the boundaries of
go's filepath libs.  one thing that made it easier was our fakefs.